sympy/combinatorics/fp_groups.py

```diff
@@ -1605,69 +1605,58 @@ def _simplification_technique_1(rels):
-    # removal can possibly happen
-    nw = [None]*l_rels
-    for i in range(l_rels):
-        w = rels[i].identity_cyclic_reduction()
-        if w.number_syllables() == 1:
-
-            # replace one syllable relator with the corresponding inverse
-            # element, for ex. x**-4 -> x**4 in relator list
-            if w.array_form[0][1] < 0:
-                rels[i] = w**-1
-            one_syllable_rels.add(rels[i])
-
-        # since modifies the array rep., so should be
-        # added a list
-        nw[i] = list(rels[i].array_form)
-
-    # bound the exponent of relators, making use of the single
+    # dictionary with "gen: n" where gen^n is one of the relators
+    exps = {}
+    for i in range(len(rels)):
+        rel = rels[i]
+        if rel.number_syllables() == 1:
+            g = rel[0]
+            exp = abs(rel.array_form[0][1])
+            if rel.array_form[0][1] < 0:
+                rels[i] = rels[i]**-1
+                g = g**-1
+                sign = -1
+            else:
+                sign = 1
+            if not g in exps or exps[g].array_form[0][1] > exp:
+                exps[g] = rel**sign
+
+    one_syllables_words = exps.values()
+    # decrease some of the exponents in relators, making use of the single
     # syllable relators
-    for i in range(l_rels):
-        k = nw[i]
-        rels_i = rels[i]
-        for gen in one_syllable_rels:
-            n = gen.array_form[0][1]
-            gen_arr0 = gen.array_form[0][0]
-            j = len(k) - 1
-            while j >= 0:
-                if gen_arr0 == k[j][0] and gen is not rels_i:
-                    t = Mod(k[j][1], n)
-
-                    # multiple of one syllable relator
-                    if t == 0:
-                        del k[j]
-                        zero_mul_simp(k, j - 1)
-                        j = len(k)
-
-                    # power should be bounded by (-n/2, n/2]
-                    elif t <= n/2:
-                        k[j] = k[j][0], Mod(k[j][1], n)
-                    elif t > n/2:
-                        k[j] = k[j][0], Mod(k[j][1], n) - n
-                j -= 1
+    for i in range(len(rels)):
+        rel = rels[i]
+        if rel in one_syllables_words:
+            continue
+        rel = rel.eliminate_words(one_syllables_words, _all = True)
+        # if rels[i] contains g**n where abs(n) is greater than half of the power p
+        # of g in exps, g**n can be replaced by g**(n-p) (or g**(p-n) if n<0)
+        for g in rel.contains_generators():
+            if g in exps:
+                exp = exps[g].array_form[0][1]
+                if exp % 2 == 0:
+                    max_exp = exp//2
+                else:
+                    max_exp = exp//2 + 1
+                rel = rel.eliminate_word(g**(max_exp), g**(max_exp-exp), _all = True)
+                rel = rel.eliminate_word(g**(-max_exp), g**(-(max_exp-exp)), _all = True)
+        rels[i] = rel
```

Comment:
There is no problem with the for loop. I was thinking of the possibility that the caller would want to do something else with original the `rels` list believing that it was intact. It may not matter currently, but the situation might change in the future.